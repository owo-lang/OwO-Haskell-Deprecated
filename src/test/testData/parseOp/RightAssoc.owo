module RightAssoc where

  infixr 1 ::
  fib = 0 :: 1 :: zipWith plus fib (tail fib)
