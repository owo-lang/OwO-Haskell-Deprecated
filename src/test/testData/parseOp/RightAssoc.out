File type: CodeFileType
Top module name: .RightAssoc
 implementation of function  "fib" (41 4 3) (44 4 6)
  no pragmas
  clause for  "fib" (41 4 3) (44 4 6)
  patterns
  function body
   function application
   function being applied
    function application
    function being applied
     named reference "::" (54 4 16) (56 4 18)
    value applied to the function
     function application
     function being applied
      function application
      function being applied
       named reference "::" (49 4 11) (51 4 13)
      value applied to the function
       constant (47 4 9) (48 4 10)
        IntegerConst 0
     value applied to the function
      constant (52 4 14) (53 4 15)
       IntegerConst 1
   value applied to the function
    function application
    function being applied
     function application
     function being applied
      function application
      function being applied
       named reference "zipWith" (57 4 19) (64 4 26)
      value applied to the function
       named reference "plus" (65 4 27) (69 4 31)
     value applied to the function
      named reference "fib" (70 4 32) (73 4 35)
    value applied to the function
     function application
     function being applied
      named reference "tail" (75 4 37) (79 4 41)
     value applied to the function
      named reference "fib" (80 4 42) (83 4 45)
  no with abstraction
  no where clause
